### PR TITLE
Added functionality for setting custom messages into current translator. Also updated documentation.

### PR DESCRIPTION
**Pull request by chrissound**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -444,8 +444,13 @@ Writing a rule class is straightforward:
 - Use `getValue()` to get the value being validated, and `setValue()` to change
   the value being sanitized.
 
-- Add a property `$message` to indicate a string that should be translated
-  as a message when validation or sanitizing fails.
+- Add a property `$message_map` as array with the below keys of which values indicate a message key that should be translated
+  as a message.
+    * failure_is
```
Comment (harikt) [inline README.md:449]: `failure_is`, `failure_is_not` etc seems good to be in capital letters.

**Comment by harikt** [comment]
The functionality seems good to me. Let @pmjones decide on this.

File: README.md (hunk 1 above)
Comment (harikt) [inline README.md:449]: sorry this was same as in https://github.com/auraphp/Aura.Filter/blob/develop/src/Aura/Filter/Rule/EqualToField.php#L34-L40  https://github.com/auraphp/Aura.Filter/blob/develop/src/Aura/Filter/Rule/EqualToValue.php#L34-L40  .

**Comment by harikt** [comment]
@cordoval this sounds good addition to v2 to be merged. 
@pmjones good for merge.

**Comment by cordoval** [comment]
i am not seeing tests added with it though

**Comment by harikt** [comment]
@pmjones can we merge this ?

**Comment by pmjones** [comment]
I don't think I have a problem with this, but the PR is against master. If you can resubmit against develop (not develop-2!) I can review it better. Sorry for the hassle, and thanks for the submission!

**Comment by harikt** [comment]
aah good catch @pmjones . I didn't notice it .

**Comment by pmjones** [comment]
Sorry to let this languish for so long. @chrissound this PR is against master, which is supposed to represent the most recent 1.x release (as vs the development work), If you can re-issue the PR against "develop" that will help.

**Comment by pmjones** [comment]
Hand-merged in https://github.com/auraphp/Aura.Filter/commit/266ad477f5c317f35a816ba7c8ba943c70280cde

**Pull request closed without merging**